code/modules/admin/verbs/one_click_antag.dm

```diff
@@ -286,69 +286,44 @@
 	return 0
 
 /datum/admins/proc/makeThunderdomeTeams() // Not strictly an antag, but this seemed to be the best place to put it.
-
-	var/list/mob/candidates = list()
-	var/mob/theghost = null
-	var/time_passed = world.time
-
+	var/max_thunderdome_players = 10
+	var/team_to_assign_to = "Green"
+	var/list/red_spawn_locations = GLOB.tdome1.Copy()
+	var/list/blue_spawn_locations = GLOB.tdome2.Copy()
 	log_admin("[key_name(owner)] tried making Thunderdome Teams with One-Click-Antag")
-	message_admins("[key_name_admin(owner)] tried making Thunderdone Teams with One-Click-Antag")
+	message_admins("[key_name_admin(owner)] tried making Thunderdome Teams with One-Click-Antag")
 
 	//Generates a list of candidates from active ghosts.
-	for(var/mob/G in GLOB.respawnable_list)
-		spawn(0)
-			switch(alert(G,"Do you wish to be considered for a Thunderdome match about to start?","Please answer in 30 seconds!","Yes","No"))
-				if("Yes")
-					if((world.time-time_passed)>300)//If more than 30 game seconds passed.
-						return
-					candidates += G
-				if("No")
-					return
-				else
-					return
-
-	sleep(300) //Debug.
-
-	for(var/mob/dead/observer/G in candidates)
-		if(!G.key)
-			candidates.Remove(G)
-
-	if(candidates.len)
-		var/teamOneMembers = 5
-		var/teamTwoMembers = 5
+	var/list/thunderdome_candidates = shuffle(SSghost_spawns.poll_candidates("Participate in a Thunderdome match?",,, 30 SECONDS, TRUE,, TRUE, FALSE))
```

Comment:
could you make `shuffle()` use named arguments past the first parameter? Blank entries with those comments is kinda bad practice

Reply:
Done